dao-voting/src/main.sw

```diff
@@ -0,0 +1,312 @@
+        require(proposal.acceptance_percentage <= acceptance_percentage, ProposalError::ApprovalPercentageNotMet);
+
+        asm(memory_address: proposal.call_data.memory_address, num_coins_to_forward: proposal.call_data.num_coins_to_forward, asset_id_of_coins_to_forward: proposal.call_data.asset_id_of_coins_to_forward, amount_of_gas_to_forward: proposal.call_data.amount_of_gas_to_forward) {
+            call memory_address num_coins_to_forward asset_id_of_coins_to_forward amount_of_gas_to_forward;
+        }
+        // Users can now convert their votes back into tokens
+        log(ExecuteEvent {
+            id: proposal_id, 
+        });
+    }
+
+    /// Unlock governance tokens which have been locked by users who have voted on a proposal
+    /// If the user had not voted on the given, expired proposal then nothing happens
+    ///
+    /// # Arguments
+    ///
+    /// - `proposal_id` - proposal to turn user votes back into governance tokens
+    ///
+    /// # Reverts
+    ///
+    /// * When the constructor has not ben called to initialize
+    /// * When the proposal id is invalid
+    /// * When the proposal is still active
+    #[storage(read, write)]fn unlock_votes(proposal_id: u64) {
+        require(proposal_id < storage.proposal_count, UserError::InvalidId);
+
+        let proposal = storage.proposals.get(proposal_id);
+        require(proposal.end_height < height(), ProposalError::ProposalStillActive);
+
+        let sender: Identity = msg_sender().unwrap();
+        let votes = storage.votes.get((sender, proposal_id));
+
+        storage.votes.insert((sender, proposal_id), 0);
+
+        storage.balances.insert(sender, storage.balances.get(sender) + votes);
+
+        log(UnlockVotesEvent {
+            id: proposal_id, vote_amount: votes, 
+        });
+    }
+
+    /// Return the amount of governance tokens in this contract
+    #[storage(read)]fn balance() -> u64 {
+        this_balance(storage.gov_token)
+    }
+
+    /// Return the amount of governance tokens a user has in this contract
+    ///
+    /// # Arguments
+    ///
+    /// - `user` - Identity to look up governance token balance in this contract.
+    #[storage(read)]fn user_balance(user: Identity) -> u64 {
+        storage.balances.get(user)
+    }
+
+    /// Return the amount of votes a user has used on a proposal
+    ///
+    /// # Arguments
+    ///
+    /// - `user` - Identity to look up votes spent on a specified proposal
+    /// - `proposal_id` - Identifier used to specifiy a proposal (0 <= proposal_id < proposal_count)
+    #[storage(read)]fn user_votes(user: Identity, proposal_id: u64) -> u64 {
+        require(proposal_id < storage.proposal_count, UserError::InvalidId);
+        storage.votes.get((user, proposal_id))
+    }
+
+    /// Return proposal data for a given id
+    ///
+    /// # Arguments
+    ///
+    /// - `proposal_id` - Identifier used to specify a proposal (0 <= proposal_id < proposal_count)
```

Comment:
You are using different descriptions for the same param. Pick 1 and use it everywhere consistently